### ISSUE TITLE
__attribute__((used)) variable causes asm.js type error in browser

### ISSUE THREAD

**Issue by duanyao** [kripken/emscripten#3424]
The test case:

```
// export_test.c
static int  __attribute__((used))  testUsed = 15234;
```

Compiling with

```
emcc -O1 export_test.c -o export_test.html
```

Load the output html by Firefox (37), an error message is shown in the console:

```
TypeError: asm.js type error: exported function name '_testUsed' not found
```

This error seems preventing the js code from being compiled as asm.js.

My emcc version is 1.31.0 (commit 01d531a5fabc58d9f83a628df5c5b43ecc5950d5).

**Comment by kripken** [comment]
This was actually fixed just a few days ago, after `1.31.0`, by pure coincidence (during work on dynamic linking, it was noticed). I verified it works on current incoming. Thanks for filing though!

**Comment by duanyao** [comment]
Great, thanks!
